lib/haml_lint/tree/tag_node.rb

```diff
@@ -20,6 +20,14 @@ def contains_script?
       @value[:parse] && !@value[:value].strip.empty?
     end
 
+    # Returns whether this tag has a specified attribute
+    # `=`).
+    #
+    # @return [true,false]
+    def has_hash_attribute?(attribute)
+      hash_attributes? && eval(hash_attributes_source).has_key?(attribute)
```

Comment:
There is no guarantee that `eval`ing the hash attributes source will work. For example, you could have dynamic variables in the hash which of course are not defined in the context of the linter `haml-lint` run. Also, [`eval` in general is dangerous](http://stackoverflow.com/a/1903032), though that goes without saying.

You need to parse this using the `parse_ruby` helper. See https://github.com/causes/haml-lint/blob/master/lib/haml_lint/linter/class_attribute_with_static_value.rb#L30 for an example.

This is pretty new ground that I haven't developed robust abstractions for (not sure if it's really possible), so you'll need to understand and interact with the abstract syntax tree generated by the `parser` gem in order to do this properly.